### PR TITLE
Prettify Initial App Template

### PR DESCRIPTION
**Pull request by JSMonk**
I changed the initial app template to show pros of the language by the simple app like it does `create-react-app`.
Screenshot of the app template:

Screen Shot 2021-01-22 at 9 59 59 PM

### CODE REVIEW

File: src/app_init/scaffold.cr [hunk 1]
```diff
@@ -0,0 +1,95 @@
+module Mint
+  class Scaffold
+    APP = "#{__DIR__}/app"
```
Comment (Sija) [inline src/app_init/scaffold.cr:3]:
```suggestion
    APP_DIR = PATH[__DIR__}, "app"]
```

File: src/app_init/scaffold.cr [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      terminal.puts "#{COG} Writing initial files:"
+
+      touch_initial_files
+      touch_config
+
+      show_created_files(path.to_s, "  ")
+    end
+
+    private def touch_initial_files
+      FileUtils.cp_r(APP, "./")
```
Comment (Sija) [inline src/app_init/scaffold.cr:39]:
```suggestion
      FileUtils.cp_r(APP_DIR.to_s, "./")
```

File: src/app_init/scaffold.cr [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        FileUtils.mkdir_p File.dirname(file_path)
+        File.write(file_path, json.to_pretty_json)
+    end
+
+    private def show_created_files(path : String, tabs : String)
+      dirs = Dir.open(path).select do |child| 
+        child != "." &&
+        child != ".." &&
+        File.directory? File.join(path, child)
+      end
```
Comment (Sija) [inline src/app_init/scaffold.cr]: Avoid using `Dir.open` without a matching call to `Dir#close`. In this case it's better to use `Dir.each_child`.

```suggestion
      dirs, files = %w[], %w[]

      Dir.each_child(path) do |child|
        child_path = File.join(path, child)
        case
        when File.directory?(child_path)
          dirs << child
        when File.file?(child_path)
          files << child
        end
      end
```

File: src/app_init/scaffold.cr [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        terminal.puts "#{tabs}#{ARROW} #{child}" 
+        show_created_files(child_path, "  #{tabs}")
+      end
+
+      files = Dir.open(path).select do |child| 
+        File.file? File.join(path, child)
+      end
+
+      files.each do |child|
+        terminal.puts "#{tabs}#{ARROW} #{child}"
```
Comment (Sija) [inline src/app_init/scaffold.cr]:
```suggestion
        terminal.puts "#{" " * indent}#{ARROW} #{child}" 
```

File: src/app_init/scaffold.cr [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      dirs = Dir.open(path).select do |child| 
+        child != "." &&
+        child != ".." &&
+        File.directory? File.join(path, child)
+      end
+
+      dirs.each do |child|
+        child_path = File.join(path, child)
+        terminal.puts "#{tabs}#{ARROW} #{child}" 
+        show_created_files(child_path, "  #{tabs}")
```
Comment (Sija) [inline src/app_init/scaffold.cr:58]:
```suggestion
        show_created_files(child_path, indent + 2)
```

File: src/app_init/scaffold.cr [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    private def show_created_files(path : String, tabs : String)
+      dirs = Dir.open(path).select do |child| 
+        child != "." &&
+        child != ".." &&
+        File.directory? File.join(path, child)
+      end
+
+      dirs.each do |child|
+        child_path = File.join(path, child)
+        terminal.puts "#{tabs}#{ARROW} #{child}"
```
Comment (Sija) [inline src/app_init/scaffold.cr]:
```suggestion
        terminal.puts "#{" " * indent}#{ARROW} #{child}" 
```

File: src/app_init/scaffold.cr [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      FileUtils.cp_r(APP, "./")
+    end
+
+    private def touch_config
+        file_path = "mint.json"
+        FileUtils.mkdir_p File.dirname(file_path)
+        File.write(file_path, json.to_pretty_json)
+    end
+
+    private def show_created_files(path : String, tabs : String)
```
Comment (Sija) [inline src/app_init/scaffold.cr:48]:
```suggestion
    private def show_created_files(path : Path, indent : Int32 = 0)
```

File: src/app_init/scaffold.cr [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+      dirs.each do |child|
+        child_path = File.join(path, child)
+        terminal.puts "#{tabs}#{ARROW} #{child}" 
+        show_created_files(child_path, "  #{tabs}")
+      end
+
+      files = Dir.open(path).select do |child| 
+        File.file? File.join(path, child)
+      end
```
Comment (Sija) [inline src/app_init/scaffold.cr]:
```suggestion
      files = %w[]
      Dir.each_child(path) do |child| 
        files << child if File.file? File.join(path, child)
      end
```

File: src/app_init/scaffold.cr [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+      FileUtils.mkdir_p path.to_s
+      FileUtils.cd path.to_s
+
+      terminal.puts "#{COG} Writing initial files:"
+
+      touch_initial_files
+      touch_config
+
+      show_created_files(path.to_s, "  ")
```
Comment (Sija) [inline src/app_init/scaffold.cr:35]:
```suggestion
      Dir.cd(path) do
        terminal.puts "#{COG} Writing initial files:"

        touch_initial_files
        touch_config

        show_created_files(path, indent: 2)
      end
```

File: src/app_init/scaffold.cr [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+      dirs.each do |child|
+        child_path = File.join(path, child)
+        terminal.puts "#{tabs}#{ARROW} #{child}"
+        show_created_files(child_path, "  #{tabs}")
+      end
+
+      files = Dir.open(path).select do |child|
+        File.file? File.join(path, child)
+      end
+
```
Comment (Sija) [inline src/app_init/scaffold.cr:64]:
```suggestion
```

File: src/app_init/scaffold.cr [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+      FileUtils.mkdir_p File.dirname(file_path)
+      File.write(file_path, json.to_pretty_json)
+    end
+
+    private def show_created_files(path : String, tabs : String)
+      dirs = Dir.open(path).select do |child|
+        child != "." &&
+          child != ".." &&
+          File.directory? File.join(path, child)
+      end
```
Comment (Sija) [inline src/app_init/scaffold.cr:53]:
```suggestion
      dirs = %w[], files = %w[]

      Dir.each_child(path) do |child|
        child_path = File.join(path, child)
        case
        when File.directory?(child_path)
          dirs << child
        when File.file?(child_path)
          files << child
        end
      end
```

**Review by Sija — COMMENTED** [review]
Great contribution 🎉  Overall it's good, I left several perf/semantic suggestions. Thanks ❤️

**Comment by JSMonk** [comment]
> Great contribution 🎉 Overall it's good, I left several perf/semantic suggestions. Thanks ❤️

Thank you 🙏 
I will fix it this Saturday.

File: src/app_init/scaffold.cr [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+      dirs.each do |child|
+        child_path = File.join(path, child)
+        terminal.puts "#{tabs}#{ARROW} #{child}"
```
Comment (Sija) [inline src/app_init/scaffold.cr:57]:
```suggestion
        terminal.puts "#{" " * indent}#{ARROW} #{child}"
```

**Comment by JSMonk** [comment]
Updated 🎉

File: src/app_init/scaffold.cr [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+      end
+
+      dirs.each do |child|
+        terminal.puts "#{tabs}#{ARROW} #{child}"
+        show_created_files(Path[path, child], indent + 2)
+      end
+
+      files.each do |child|
+        terminal.puts "#{tabs}#{ARROW} #{child}"
+      end
```
Comment (Sija) [inline src/app_init/scaffold.cr:71]:
```suggestion
      dirs = %w[]
      files = %w[]

      Dir.each_child(path) do |child|
        child_path = File.join(path, child)
        case
        when File.directory?(child_path)
          dirs << child
        when File.file?(child_path)
          files << child
        end
      end

      marker = "#(" " * indent}#{ARROW} "

      dirs.each do |child|
        terminal << marker << child << '\n'
        show_created_files(Path[path, child], indent + 2)
      end

      files.each do |child|
        terminal << marker << child << '\n'
      end
```

**Review by Sija — COMMENTED** [review]
One small suggestion and it's GTG.

File: src/app_init/scaffold.cr (hunk 13 above)
Comment (JSMonk) [reply src/app_init/scaffold.cr]: Done ✅

**Review by Sija — APPROVED** [review]
LGTM 🚀

**Comment by gdotdesign** [comment]
Thanks! :tada:

**Pull request merged**